src/vs/gitpod/browser/workbench/workbench-dev.html

```diff
@@ -2,6 +2,76 @@
 <!DOCTYPE html>
 <html>
 	<head>
+		<script>
+			async function gitpodMetricsAddCounter(metricsName, labels, value) {
+				function getMetricsUrl() {
+					const gitpodHost = "{{GITPOD_HOST}}";
+					if (!gitpodHost) {
+						return '';
+					}
+					return `https://ide.${gitpodHost}/metrics-api`;
+				}
+				try {
+					const metricsUrl = getMetricsUrl();
+					if (!metricsUrl) {
+						return false;
+					}
+					const url = `${metricsUrl}/metrics/counter/add/${metricsName}`;
+					const params = { value, labels };
+					const response = await fetch(url, {
+						method: 'POST',
+						body: JSON.stringify(params),
+						credentials: 'omit',
+					});
+					if (!response.ok) {
+						const data = await response.json(); // { code: number; message: string; }
+						console.error(`Cannot report metrics with addCounter: ${response.status} ${response.statusText}`, data);
+						return false;
+					}
+					return true;
+				} catch (err) {
+					console.error('Cannot report metrics with addCounter, error:', err);
+					return false;
+				}
+			}
+
+			// sum(rate(gitpod_vscode_web_load_total{status='failed'}[2m]))/sum(rate(gitpod_vscode_web_load_total{status='loading'}[2m]))
+			const gitpodVSCodeWebLoadTotal = 'gitpod_vscode_web_load_total';
+
+			gitpodMetricsAddCounter(gitpodVSCodeWebLoadTotal, { status: 'loading' });
+			gitpodMetricsAddCounter('gitpod_supervisor_frontend_client_total');
+
+			let vscodeWebFailedToLoad = false;
+			const onVsCodeWorkbenchError = (event) => {
+				if (!vscodeWebFailedToLoad) {
+					vscodeWebFailedToLoad = true;
+					gitpodMetricsAddCounter(gitpodVSCodeWebLoadTotal, { status: 'failed' });
+				}
+
+				// If the error is not an element loading error, we only increase the total error count
+				if (typeof event?.target?.getAttribute !== 'function') {
+					gitpodMetricsAddCounter('gitpod_supervisor_frontend_error_total');
+					return;
+				}
+				const labels = {};
+
+				// We take a look at the resource that was attempted to load
+				const resourceSource = event.target.getAttribute('src') || event.target.getAttribute('href');
+
+				if (resourceSource) {
+					if (resourceSource.match(new RegExp(/\/build\/ide\/code:.+\/__files__\//g))) {
+						// TODO(ak) reconsider how to hide knowledge of VS Code from supervisor frontend, i.e instrument amd loader instead
+						labels['resource'] = 'vscode-web-workbench';
+					}
```

Comment:
I wonder if this check is relevant here, because we are checking for errors in the whole window and there may be resources like in `<iframe>`s which may trigger this, so the check is perhaps justified here.